payjoin/tests/integration.rs

```diff
@@ -1,3 +1,4 @@
+#[cfg(all(feature = "v1", feature = "v2"))]
```

Comment:
This would remove the ability to run integration tests for `v2` only (or for `v1` only).

```suggestion
#[cfg(any(feature = "v1", feature = "v2"))]
```

Actually, this is probably the reason for the drastic coverage reduction.

Reply:
Hmm. I changed this because we're depending on `payjoin::receive::v1::build_v1_pj_uri` for both v1 and v2 tests. Why do you think this would reduce coverage? Because some runs would not check branches that they previously had, even though all of the same lines are still covered? That doesn't make sense to me.

Reply:
I see now, our coverage script:

```
cargo llvm-cov --no-report --no-default-features --features=v1,_danger-local-https
cargo llvm-cov --no-report --no-default-features --features=v2,_danger-local-https,io
```

Doesn't test the two in tandem since the integration/v2 test module now depends on the v1 feature since there are many v2 tests that test backwards compatibility.

Reply:
`contrib/coverage.sh` runs the tests like this:

```
cargo llvm-cov --no-report --no-default-features --features=v1,_danger-local-https
cargo llvm-cov --no-report --no-default-features --features=v2,_danger-local-https,io
```

Default features are disabled, so the first command wouldn't pass the `cfg` check. 

Though now that features are additive maybe these two commands can be combined with `--all-features`.

(The same applies to lint and test scripts)

Reply:
jinx